### PR TITLE
Adding PrepEnrollmentDate to the Fact and the Linelist

### PR DESCRIPTION
**Pull request by Marymary-dev**

### CODE REVIEW

File: Scripts/NDWH/PREP_FACT_TABLES/load_FactPrepAssessments.sql [hunk 1]
```diff
@@ -82,6 +84,8 @@ left join NDWH.dbo.DimAgency as agency on agency.AgencyName = MFL_partner_agency
 left join NDWH.dbo.DimFacility as facility on facility.MFLCode = source_data.SiteCode
 left join NDWH.dbo.DimAgeGroup as age_group on age_group.Age = datediff(yy, patient.DOB, coalesce(source_data.AssessmentVisitDate, getdate()))
 left join NDWH.dbo.DimDate as assessment_date on assessment_date.Date = source_data.AssessmentVisitDate
+left join NDWH.dbo.DimDate as prep_enrol_Date on prep_enrol_Date.Date = source_data.PrepEnrollmentDate
```
Comment (nobert-mumo) [inline Scripts/NDWH/PREP_FACT_TABLES/load_FactPrepAssessments.sql:87]: @Marymary-dev I know we have a similar field on DimPatient. Would this be any different ?

**Comment by Marymary-dev** [comment]
is it, let me check that

File: Scripts/REPORTING/load_Linelist_Prep.sql [hunk 2]
```diff
@@ -19,7 +19,8 @@ WITH prepCascade AS  (
         ass.year AssessmentYear,
         EOMONTH(ass.Date) as AsofDate,
         EligiblePrep,
-        ScreenedPrep
+        ScreenedPrep,
+        prepEnrol.DateKey as PrepEnrollmentDate
```
Comment (nobert-mumo) [inline Scripts/REPORTING/load_Linelist_Prep.sql:23]: @Marymary-dev I thought the idea behind join to the date dimension is to pick the corresponding date field ?

Comment (nobert-mumo) [reply Scripts/REPORTING/load_Linelist_Prep.sql:23]: @Marymary-dev  I think if you pick `prepEnrol.Date` here then we don't have to cast to date down there.

**Review by nobert-mumo — CHANGES_REQUESTED** [review]

File: Scripts/REPORTING/load_Linelist_Prep.sql (hunk 2 above)
Comment (Marymary-dev) [reply Scripts/REPORTING/load_Linelist_Prep.sql]: > @Marymary-dev I think if you pick `prepEnrol.Date` here then we don't have to cast to date down there.

@nobert-mumo  this has been corrected

File: Scripts/REPORTING/load_Linelist_Prep.sql [hunk 3]
```diff
@@ -19,7 +19,8 @@ WITH prepCascade AS  (
         ass.year AssessmentYear,
         EOMONTH(ass.Date) as AsofDate,
         EligiblePrep,
-        ScreenedPrep
+        ScreenedPrep,
+        cast (prepEnrol.DateKey as date) as PrepEnrollmentDate
```
Comment (nobert-mumo) [inline Scripts/REPORTING/load_Linelist_Prep.sql:23]: @Marymary-dev rem to change to `prepEnrol.Date as PrepEnrollmentDate`

**Review by nobert-mumo — CHANGES_REQUESTED** [review]

File: Scripts/REPORTING/load_Linelist_Prep.sql (hunk 3 above)
Comment (Marymary-dev) [reply Scripts/REPORTING/load_Linelist_Prep.sql]: Okay, this has been picked @nobert-mumo

**Pull request merged**